docs/rfcs/rfc-040-relay-first-api.md

```diff
@@ -0,0 +1,1282 @@
+
+Following Relay spec, the fields `last` and `before` can be used instead of `first` and `after` to do backwards pagination.
+
+### Aggregation
+
+Aggregations have been moved to be part of the projection on connection queries:
+
+```graphql
+query ShortestMovieTitleAndCount {
+    moviesConnection {
+        aggregation {
+            node {
+                title {
+                    shortest
+                }
+                count
+            }
+        }
+    }
+}
+```
+
+This means that a projection can be returned along with the edges. Nested aggregations can be queried in the same fashion. `*Aggregation` types returned in a relationship will have the `node` and `fields` properties with the respective aggregation:
+
+```graphql
+query AggregateActorsPerMovie {
+    moviesConnection {
+        edges {
+            node {
+                actors {
+                    aggregation {
+                        node {
+                            count
+                        }
+                        fields {
+                            year {
+                                min
+                                max
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+> Aggregations are not available for 1-1 relationships
+
+### Fulltext
+
+If fulltext is enable for an entity using the `@fulltext` directive, the `fulltext` input will be available, as well as the `score` fields and filters as a `Float`:
+
+```graphql
+query Query {
+    moviesConnection(
+        fulltext: { phrase: "Test", index: MovieTitle }
+        where: { edges: { fulltext: { score: { gt: 5 } } } }
+    ) {
+        totalCount
+        edges {
+            fulltext {
+                score
+            }
+        }
+    }
+}
+```
+
+> If there is a single index, the index field could be made optional ignored
+
+This approach allows the possibility of having nested fulltext:
+
+```graphql
+query Movies {
+    peopleMoviesWithFulltext {
+        edges {
+            node {
+                movies(
+                    fulltext: { phrase: "Test", index: MovieTitle }
+                    where: { edges: { fulltext: { score: { gt: 5 } } } }
+                ) {
+                    edges {
+                        fulltext {
+                            score
+                        }
+                        node {
+                            title
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+Fulltext can be sorted as a normal field:
+
+```graphql
+query TopLevelFulltext {
+    moviesConnection(
+        fulltext: { phrase: "Matrix", index: MovieTitle }
+        sort: { edges: { fulltext: { score: DESC } } }
+    ) {
+        edges {
+            node {
+                title
+            }
+        }
+    }
+}
+```
+
+> Note that the plain `fulltext` option without `where` may still filter the results.
+
```

Comment:
It would also be interesting to see an example of `fulltext` along with aggregations as we've had a number of requests for that e.g. #2681

Reply:
I think there is already an example of this case:
```graphql
query TopLevelFulltextAndAggregation {
    moviesConnection(
        fulltext: { phrase: "Matrix", index: MovieTitle }
        where: { edges: { node: { released: { eq: 2001 } }, fulltext: { score: { gt: 6 } } } }
        sort: { edges: { fulltext: { score: DESC } } }
        first: 10
    ) {
        edges {
            node {
                title
            }
            fulltext {
                score
            }
        }
        aggregation {
            node {
                count
            }
        }
    }
}
```

If you are asking about aggregation as a filter, it works the same as any filter for nested aggregations, in this example, filtering on movies with >= 10 actors

```graphql
query TopLevelFulltextAndAggregation {
  moviesConnection(
    fulltext: { phrase: "Matrix", index: MovieTitle }
    where: {
      edges: {
        node: { actors: { aggregation: { count: { gte: 5 } } } }
        fulltext: { score: { gt: 6 } }
      }
    }
    sort: { edges: { fulltext: { score: DESC } } }
    first: 10
  ) {
    edges {
      node {
        title
      }
      fulltext {
        score
      }
    }
    aggregation {
      node {
        count
      }
    }
  }
}
```